### ISSUE TITLE
regression: translate-c output struct depends on itself (false positive)

### ISSUE THREAD

**Issue by acgaudette** [ziglang/zig#17227]
### Zig Version

0.12.0-dev.464+a63a1c5cb

### Steps to Reproduce and Observed Behavior

a project depending on several C headers previously compiled, but now throws an error inside a file generated by translate-c.

the header in question is generated by cimgui (including as a keyword in case anyone else has this specific problem) and should probably be considered 'valid' as it has many dependents in the wild.

I've pared the error down to a minimal example:
```zig
const c = @cImport(@cInclude("include.h"));

pub fn main() void
{
        var b: c.B = undefined;
        _ = b;
}
```

'include.h'
```c
struct A
{
        struct C* c;
};

struct B
{
        struct A* a;
};

struct C
{
        struct B b;
};
```

```
error: struct '.cimport.struct_B' depends on itself
pub const struct_B = extern struct {
                     ~~~~~~~^~~~~~
```

the error is pretty easy to reproduce with the actual cimgui header, so I can include that if necessary.

this occurs in `0.12.0-dev.464+a63a1c5cb` but not `0.12.0-dev.415+5af5d87ad`, which means this occurred somewhere inside the big batch of commits in the last couple days.

perhaps
f91ff9a746ff97945dacc8114c073bd279f68f17 @jayschwa 
or changes by @andrewrk, @jacobly0

### Expected Behavior

compilation succeeds

**Comment by acgaudette** [comment]
to reproduce with the cimgui header:

include https://github.com/cimgui/cimgui/blob/docking_inter/cimgui.h

```zig
const c = @cImport({
        @cDefine("CIMGUI_DEFINE_ENUMS_AND_STRUCTS", {});
        @cInclude("include.h");
});

pub fn main() void
{
        var cmd: c.ImVector_ImDrawCmd = undefined;
        _ = cmd;
}
```
```
cimport.zig:552:46: error: struct '.cimport.struct_ImVector_ImDrawCmd' depends on itself
pub const struct_ImVector_ImDrawCmd = extern struct {
                                      ~~~~~~~^~~~~~
```

**Comment by jayschwa** [comment]
There are several open issues for false dependency loops, which might be getting triggered here. I think what I'll do is make a new PR that temporarily disables zero values for structs. Once those false dependency loop issues have been fixed, zero values for structs can be re-enabled. Thanks for the report.

**Comment by acgaudette** [comment]
:+1: can you link me to any of those issues?

**Comment by nektro** [comment]
https://github.com/ziglang/zig/issues?q=is%3Aissue+is%3Aopen+%22depends+on+itself%22

**Comment by jayschwa** [comment]
> 👍 can you link me to any of those issues?

#12325 and #14147 seem like the relevant ones.

**Comment by acgaudette** [comment]
ah I see. didn't expect anything so old / before `0.12.0-dev.415+5af5d87ad`

**Comment by michal-z** [comment]
This is a new bug (regression). cimgui header was working fine with `cImport`. [zig-gamedev project](https://github.com/michal-z/zig-gamedev) is also affected by this issue.

**Comment by mlugg** [comment]
Nope, it's what @jayschwa said - default zero values causing false dependency loops. That PR wasn't me.

HOWEVER! This should actually be fixed on master as of the merging of #17172 (it works according to my local tests). The issue was related to struct alignment resolution forcing full layout resolution: when `zeroes` attempted to return a pointer, it at some point tried to resolve the pointer's alignment, which attempted to resolve the layout of the struct within, which triggered a dependency loop. This was fixed by the PR linked above, because struct alignment resolution is now distinct from general layout resolution.

For anyone who experienced this bug, try your code out now - it should be fixed!

**Comment by michal-z** [comment]
I see, thanks for the info!

**Comment by jayschwa** [comment]
Confirmed that this is fixed in master after #17172 was merged. I tested with both the small reproducer and the cimgui example. It should be noted that the ziglang.org download page doesn't have a new build with the fix yet, but probably will within the next day or so.
